### ISSUE TITLE
"git log" hangs when resizing cmd.exe window

### ISSUE THREAD

**Issue by ThomasFWise** [microsoft/git#549]
- [ ] I was not able to find an [open](https://github.com/microsoft/git/issues?q=is%3Aopen)
        or [closed](https://github.com/microsoft/git/issues?q=is%3Aclosed) issue matching
        what I'm seeing, including in [the `git-for-windows/git` tracker](https://github.com/git-for-windows/git/issues).

### Setup

 - Which version of `microsoft/git` are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.38.1.vfs.0.0
cpu: x86_64
built from commit: 0de34d54680002380b4e1adad74db64a34e03053
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

Are you using Scalar or VFS for Git?

Scalar

If VFS for Git, then what version?

```
$ gvfs version

'gvfs' is not recognized as an internal or external command,
operable program or batch file.
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?
Windows 11 22H2, 64-bit

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.22621.819]
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

No. This issue repros on both a scalar and standard git repo.

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

cmd.exe. Issue repros on both classic cmd and new Windows Terminal

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
1) Run "git log"
2) Resize the window while git log is running
3) quit out of git log
```
 - What did you expect to occur after running these commands?

git.exe should have closed and allowed running further commands

 - What actually happened instead?

git.exe hangs

 - If the problem was occurring with a specific repository, can you specify
   the repository?

   * [ ] Public repo: **insert URL here**
   * [ ] Windows monorepo
   * [ X ] Office monorepo
   * [ X ] Other Microsoft-internal repo: Office/OE/OBOE
   * [ X ] Other internal repo.

**Comment by dscho** [comment]
This sounds like it refers to the same underlying bug as https://github.com/git-for-windows/git/issues/4060, and that in turn looks like it is a bug in the Cygwin/MSYS2 runtime. I asked the reporter in the ticket to report it to Cygwin, we will hopefully get some results soon. Alternatively, feel free to reproduce the problem using [Cygwin](https://cygwin.com/) and then [report the bug there](https://cygwin.com/problems.html) for hopefully expedient processing.

**Comment by garretwilson** [comment]
> I asked the reporter in the ticket to report it to Cygwin …

… which was not the appropriate course of action because I don't have the expertise to know whether your project is using Cygwin, which parts of Cygwin is being used, or how Cygwin is being used.

I updated https://github.com/git-for-windows/git/issues/4060 to indicate that the appropriate course of action would be for 1) Git for Windows to verify that the issue occurs by reproducing it, and 2) based upon the expertise of the Git for Windows developers, file a bug if needed against Cygwin or whatever project is causing the problem transitively.

As a user I have no access or special knowledge of these transitive dependencies, and I'm not the correct person to be filing bugs against those packages. I filed a bug against Git for Windows, and the Git for Windows developers need to investigate and file bugs to dependency projects as appropriate.
